media_driver/agnostic/media_srcs.cmake

```diff
@@ -85,8 +88,103 @@ if (BUILD_KERNELS)
             COMMAND KernelBinToSource -i ${krn}.bin -o ${krn_dir}/)
     endfunction()
 
+    # This function describes object files generated by cmc from the given input cm-file.
+    # If cm-file has changed, it may be required to adjust this function.
+    function(get_cm_dat_objs file objs)
+        get_filename_component(name ${src} NAME)
+        if(name STREQUAL "downscale_kernel_genx.cpp")
+            set(objs
+                downscale_kernel_genx_0.dat
+                downscale_kernel_genx_1.dat
+                PARENT_SCOPE)
+        elseif(name STREQUAL "hme_kernel_genx.cpp")
+            set(objs
+                hme_kernel_genx_0.dat
+                hme_kernel_genx_1.dat
+                hme_kernel_genx_2.dat
+                PARENT_SCOPE)
+        elseif(name STREQUAL "downscale_convert_kernel_genx.cpp")
+            set(objs
+                downscale_convert_kernel_genx_0.dat
+                PARENT_SCOPE)
+        else()
+            set(objs "" PARENT_SCOPE)
+        endif()
+    endfunction()
+
+    # Function parses the given c-file and extracts the value from the defined macro.
+    # Parser expects the first occurence in the following format:
+    #   "#define name xxx" - whitespaces are important!!
+    function(get_c_macro_int file name value)
+        file(STRINGS ${file} value_str REGEX "#define ${name}" LIMIT_COUNT 1)
+        if(value_str STREQUAL "") # old style version
+            message(FATAL_ERROR "Failed to find macro ${name} in the file: ${file}")
+        endif()
+        string(REPLACE "#define ${name} " "" value_str ${value_str})
+        set(${value} ${value_str} PARENT_SCOPE)
+    endfunction()
+
+    # Function generates kernel for the specified platform. It assumes that generated
+    # kernel should be placed in the certain directory (see ${krn_dir}).
+    function(gen_kernel_from_cm name platform index sources)
+        platform_to_genx(${platform} genx kind)
+
+        set(krn ig${name}krn_g${genx})
+        set(krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/codec/kernel_free)
+        set(out_dir "${CMAKE_CURRENT_BINARY_DIR}/kernels/codec/${platform}")
+
+        add_custom_command(
+            OUTPUT ${out_dir}
+            COMMAND ${CMAKE_COMMAND} -E make_directory ${out_dir}
+            COMMENT "Create codec kernels output directory: ${out_dir}")
+
+        # Compiling all the given sources
+        set(dats "")
+        if(genx STREQUAL "11")
+          set(genx "11lp")
```

Comment:
@cchunbo : I am **forced** to do this to match the generated kernel with your pregenerated one! As soon as I program honest -Qxcm_jit_target=gen11, resulted kernel binaries don't match. What's going on and for which platform we really generate the kernel?